### ISSUE TITLE
submenu item is not added to admin menu

### ISSUE THREAD

**Issue by kaidohallik** [hipster-labs/generator-jhipster-entity-audit#49]
If using `angular1` as `clientFramework` then generator gives error:
```
Unable to find src/main/webapp/app/layouts/navbar/navbar.component.html or missing required jhipster-needle. Reference to entity-audit not added to admin menu.
```
Problem is caused by `this.clientFramework` is undefined in main generator function `addElementToAdminMenu`.

Resolution is to use similar method like is used with `enableTranslation`: add new parameter `clientFramework` to `addElementToAdminMenu`.

Needs change in the main generator and in the entity audit module.

I can do PR-s to both projects.

**Comment by deepu105** [comment]
I think it only needs change in this module to set the clientFramework var
on the context. Ofcourse you can do PR


Thanks & Regards,
Deepu

**Comment by kaidohallik** [comment]
After some testing I found that variable `this.clientFramework` is available only if running from main generator (for example `yo jhipster:entity`).

If running from module (for example `yo jhipster-entity-audit`) then `this.clientFramework` is `undefined`.

After inspecting source code I saw that same problem was probably with other options from `.yo-rc.json` (for example `enableTranslation`) and solution was to drive these to modules and then give back as function parameters.

I made 2 PR-s with similar solution for `clientFramework`. First PR for main generator and second PR for entity audit module.

If some other module is using some function from this list:
* addElementToMenu
* addElementToAdminMenu
* addEntityToMenu
* addSocialButton

then this module doesn't work with current master and after merging main generator changes must make similar changes as are made in entity audit module.

I someone finds better solution then feel free to close these PR-s and apply better solution.
